### PR TITLE
Add high level building block diagram with mermaid JS

### PR DESCRIPTION
**Pull request by addshore**

### CODE REVIEW

**Review by Silvan-WMDE — COMMENTED** [review]
One minor comment on the diagram.

I also tested the diagram generation locally and it works.

File: diagrams/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Build a container to build the diagrams in:
+
+```lang=sh
+docker build -t wikibase-architecture-diagrams .
+```
+
+Generate a SVG representation of the mermaid diagram with:
+
+```lang=sh
```
Comment (tzhelyazkova) [inline diagrams/README.md:23]:
```suggestion
```sh
```

File: diagrams/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## [mermaidjs](https://github.com/mermaid-js/mermaid)
+
+You can use this tool for live editing: https://mermaid-js.github.io/mermaid-live-editor
+
+### Generating
+
+Fetch the needed dependencies:
+
+```lang=sh
```
Comment (tzhelyazkova) [inline diagrams/README.md:11]:
```suggestion
```sh
```

File: diagrams/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+docker run -it --rm -v "$PWD:/app" -w /app  node:14 npm install
+```
+
+Build a container to build the diagrams in:
+
+```lang=sh
```
Comment (tzhelyazkova) [inline diagrams/README.md:17]:
```suggestion
```sh
```

File: diagrams/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+Generate a SVG representation of the mermaid diagram with:
+
+```lang=sh
+docker run -it --rm -v "$PWD:/app" wikibase-architecture-diagrams node_modules/.bin/mmdc -i 05-blocks-1.mmd -o 05-blocks-1.svg
+```
+
+If you get a Chrome error `Running as root without --no-sandbox is not supported` you might need to use the `--no-sandbox` option of the Puppeteer:
+
+```lang=sh
```
Comment (tzhelyazkova) [inline diagrams/README.md:29]:
```suggestion
```sh
```

File: diagrams/05-blocks-1.mmd [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    query-ui[Queryservice UI]
+    query[Queryservice]:::notus
+    elastic[Elasticsearch]:::notus
+    commons[Wikimedia Commons]:::notus
+    quickstatements[Quickstatements]:::notus
+
+    user-wd-->|Interacts with|mwwb
+    user-quickstatements-->|Interacts with|quickstatements
+    user-query-->|Interacts with|query-ui
+    user-query-->|Queries|query
```
[the remark this answers is not in the document]
Comment (addshore) [reply diagrams/05-blocks-1.mmd:20]: Yes, as users can query the API directly, or they can use the UI

**Comment by addshore** [comment]
Generally I'm happy with the direction this took for now.
I did look at investigating "watch" stuff a little bit but decided after a few mins it probably isn't worth it yet.
I think using mermaid will make sense and that probably won't change in the future?
We will probably end up using some generator that does more things in the future (I had a quick look at githubify which has built in mermaid support in markdown files for example)

File: diagrams/README.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Generate SVGs from diagrams on local system:
+
+```sh
+npm run build:diagrams
+```
+
+OR build them in a docker container:
+
+```sh
+npm run buildd:diagrams
```
Comment (Silvan-WMDE) [inline diagrams/README.md:30]:
```suggestion
npm run build-docker:diagrams
```

File: diagrams/package.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "diagrams",
+  "scripts": {
+    "build:diagrams" : "node_modules/.bin/mmdc -p puppeteer-config.json -i 05-blocks-1.mmd -o 05-blocks-1.svg",
+    "buildd:diagrams" : "docker run --rm -it -v \"$PWD:/app\" $(docker build -t wikibase-architecture-diagrams -q .) npm run build:diagrams"
```
Comment (Silvan-WMDE) [inline diagrams/package.json:5]:
```suggestion
    "build-docker:diagrams" : "docker run --rm -it -v \"$PWD:/app\" $(docker build -t wikibase-architecture-diagrams -q .) npm run build:diagrams"
```

**Review by Silvan-WMDE — CHANGES_REQUESTED** [review]

**Review by tzhelyazkova — APPROVED** [review]

**Comment by addshore** [comment]
Regarding the diagram in general I feel that we should think about these points a bit and see if this is the approach that we want to take for documentation, or having something a bit more manual?
 - As much as manipulation can break a message, bad visuals can unintentionally spread confusion
- Entire systems can be built subtly wrong due to bad graphs or unclear but technically correct docs

Perhaps we end up with a mix of things like mermaid for sequence diagrams but some other tooling for these less structured diagrams.
I have to say that currently the concept board version of this seems a little more parsable to me.

**Comment by Silvan-WMDE** [comment]
> I have to say that currently the concept board version of this seems a little more parsable to me.

I agree about the readability of the manually created one, but we should ensure to have an SVG version of it, which conceptboard doesn't offer as export, I think.

**Review by Silvan-WMDE — APPROVED** [review]

**Pull request merged**